common/authen.js

```diff
@@ -240,49 +268,78 @@
             logInHelper(loginWindowCb, win, reloadCb, 'popUp', null, logAction);
         };
 
-        var shouldReloadPageAfterLogin = function(newSession) {
-            if (_session === null) return true;
+        var shouldReloadPageAfterLogin = function() {
+            if (_session === null && _prevSession == null) return true;
             return false;
         };
 
-        return {
+        /**
+         * Will return a promise that is resolved with the session.
+         * It will also call the _executeListeners() functions and sets the _session.
+         * If we couldn't fetch the session, it will resolve with `null`.
+         *
+         * @param  {string=} context undefined or "401"
+         */
+        var _getSession = function(context) {
+            var config = {
+                skipHTTP401Handling: true,
+                headers: {}
+            };
 
-            /**
-             * Will return a promise that is resolved with the session.
-             * It will also call the _executeListeners() functions and sets the _session.
-             * If we couldn't fetch the session, it will resolve with `null`.
-             *
-             * TODO needs to be revisited for 401.
-             * We want to reload the page and stop the code execution for 401,
-             *  but currently the code will continue executing and then reloads.
-             *
-             * @param  {string=} context undefined or "401"
-             */
-            getSession: function(context) {
-                var config = {
-                    skipHTTP401Handling: true,
-                    headers: {}
-                };
-                config.headers[ERMrest.contextHeaderName] = {
-                    action: logService.getActionString(logService.logActions.SESSION_RETRIEVE, "", "")
+            config.headers[ERMrest.contextHeaderName] = {
+                action: logService.getActionString(logService.logActions.SESSION_RETRIEVE, "", "")
+            }
+
+            // see if there's a token before doing any decision making
+
+            // if there's no webauthnCookie || no cookieFromStorage, login flow
+
+            // if there's a webauthnCookie user was logged in at some point on this machine, check to see if it matches _cookie
```

Comment:
The code is not doing any of the explained comment here, right? Is this comment supposed to explain how you want to implement it later? I so, I don't think it should be part of the code.

Reply:
From above:

> commented logic path for managing timeouts in the future
started this while we were discussing implementing #1916 and #1918